lib/spoom/sorbet/sigils.rb

```diff
@@ -84,13 +84,49 @@ def self.change_sigil_in_files(path_list, new_strictness)
         params(
           directory: T.any(String, Pathname),
           strictness: String,
-          extension: String
+          extension: String,
+          desired: String,
+          all: T::Boolean,
+          below: T.nilable(String),
+          above: T.nilable(String)
         ).returns(T::Array[String])
       end
-      def self.files_with_sigil_strictness(directory, strictness, extension: ".rb")
+      def self.files_with_sigil_strictness(directory, strictness, extension: ".rb",
+        desired: "true", all: false, below: nil, above: nil)
```

Comment:
I think it would be easier if this method was always expecting an array of strictnesses to select the files matching.

Something like:
```rb
      # finds all files in the specified directory matching `strictnesses`
      sig { params(directory: T.any(String, Pathname), strictness: T::Array[String], extension: String).returns(T::Array[String]) }
      def self.files_with_matching_strictness(directory, strictnesses, extension: ".rb")
        paths = Dir.glob("#{File.expand_path(directory)}/**/*#{extension}").sort.uniq
        paths.filter do |path|
          strictnesses.include?(file_strictness(path))
        end
      end
```

Reply:
I don't think I agree with this one, because then that would require figuring out the range of sigils that above/below/all translate into in the `bump.rb` in order to pass them to the method. I feel like it makes more sense for that logic to live and be handled here so that `bump.rb` is just passing the options given in the input, and `sigils.rb` is translating that into what is actually needed.

Reply:
I'm not sure I understand since you can still use the method `sigil_range` from below:

```rb
strictnesses = Sigils.sigils_range(desired, below: "strong", above: "ignore", all: false)
files_to_bump = Sigils. files_with_matching_strictness(dir, strictnesses)
```